staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -113,6 +113,20 @@ type Config struct {
 	// Version forces a specific version to be used (if registered)
 	// Do we need this?
 	// Version string
+
+	// BearerTokenSource for managing the token
+	BearerTokenSource BearerTokenSource
```

Comment:
`func (c *Config) TransportConfig()` sets up WrapTransport from the AuthProvider config. If there is already a WrapTransport provided, it composes them together (the one in the config struct and the one from the auth provider).

This is adding two ways to obtain a BearerTokenSource (providing one directly in the config struct or by setting an auth provider config), but does not compose the two methods together (or describe how those would interact if you provided both)